### ISSUE TITLE
Automatically reload configuration

### ISSUE THREAD

**Issue by jhernand** [openshift/autoheal#41]
Currently the auto-heal service loads the configuration only when it is started. If any changes happen to the configuration file or to the Kubernetes secrets that it uses then the service needs to be restarted. Instead of that the service should watch for changes in the configuration file and in the secrets and reload them automatically. This watching should be implemented internally in the `Config` type. Other parts of the service should have a mechanism to be notified of changes in the configuration, for example:

```go
type ConfigChangeEvent {
}

type ConfigChangeListener func (*ConfigChangeEvent)

config := ...
awxAddress := config.AWX().Address()
config.AddConfigChangeListener(func () {
  awsAddress = config.AWX().Address()
})
```

**Comment by jhernand** [comment]
I assigned this to myself because I can't add @yaacov, but he is the real assignee.

**Comment by yaacov** [comment]
Thanks

**Comment by yaacov** [comment]
closed by #97
